### DISCUSSION TITLE
Mocking Feature in DGS

### DISCUSSION THREAD

**Discussion by jmbarroso** [Netflix/dgs-framework#156] [Q&A] [answered]
Hi all,

I found in the advance section an entry dedicated to [Mocking](https://netflix.github.io/dgs/advanced/mocking/). I have a question related to enabling it remotely. The documentation says:

>  but the true power comes from enabling the mocks remotely

I didn't see more detail about how to enable them. Does it mean that I can have a mock imp and a real one and that remotely I can enable one or the other? 

Thanks

**Comment by paulbakker** [comment, marked as answer]
This is slightly embarrassing, but that future doesn't actually exist...
We used to have a feature like this based on an internal testing system, but actually got rid of it because it wasn't used much.
So that "true power" turned out to be not that powerful after all. 

I'll remove the text, and sorry for the disappointment!

**Reply by jmbarroso** [reply]
Thanks, Paul. And no worries about that. The framework is still awesome even without the remote mock switch. :)

I have two follow-up questions if you don't mind. Based on what I see, the Mock helps to skip the data fetcher implementation so you can serve mock data straight from the map. So two questions: 
1) is my understanding correct? or is there any other Mock usage beyond providing a shortcut to serve data?
2) fields in the Mock Map seem to override the Data-Fetcher. Any reason not to be the other way around (when no data-fetcher use the map to fill in the request?)

**Reply by paulbakker** [reply]
1. Correct
2. This way you can also use it to override a real datafetcher to force a certain result for testing. E.g. you could use this with some custom code to provide values remotely.

**Reply by jmbarroso** [reply]
Thanks for the answer
